### PR TITLE
General Madrigal Support 

### PR DESCRIPTION
**Pull request by rstoneback**
Bonus JRO-ISR typo correction.

General madrigal instrument file needs small modification to include list file support. We could require the user to supply an input template string when they instantiate the Instrument object, along with the Madrigal instrument and kindat code.

### CODE REVIEW

File: pysat/instruments/jro_isr.py [hunk 1]
```diff
@@ -60,7 +60,7 @@
                       'drifts_ave':jro_fname1 + "drifts_ave" + jro_fname2,
                       'oblique_stan':jro_fname1 + jro_fname2,
                       'oblique_rand':jro_fname1 + "?" + jro_fname2,
-                      'oblique_rand':jro_fname1 + "?" + jro_fname2}
+                      'oblique_long':jro_fname1 + "?" + jro_fname2}
```
Comment (aburrell) [inline pysat/instruments/jro_isr.py]: Nice catch

File: pysat/instruments/madrigal_pandas.py [hunk 2]
```diff
@@ -0,0 +1,187 @@
+# -*- coding: utf-8 -*-.
+"""Supports generalized access to Madrigal Data.
+
+To use this routine, you need to know both the Madrigal Instrument code
+as well as the data tag numbers that Madrigal uses to uniquely identify
+data sets. Using these codes, the madrigal_methods.py routines will
+be used to support downloading and loading of data.
+
+
+Downloads data from the JRO Madrigal Database.
```
Comment (aburrell) [inline pysat/instruments/madrigal_pandas.py]: Do we want to default to the JRO database?

File: pysat/instruments/madrigal_pandas.py [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+# support load routine
+load = mad_meth.load
+
+# support download routine
+# real download attached during init
+# however, pysat requires a method before we get there
+download = mad_meth.download
+
+def init(self):
+    """Initializes the Instrument object with values specific to JRO ISR
```
Comment (aburrell) [inline pysat/instruments/madrigal_pandas.py]: Looks like an old docstring

File: pysat/instruments/madrigal_pandas.py [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+
+    """
+    mad_meth.download(date_array, inst_code=inst_code,
+                      kindat=kindat,
+                      data_path=data_path, user=user, password=password)
+
+   
+        
+def clean(self):
+    """Routine to return JRO ISR data cleaned to the specified level
```
Comment (aburrell) [inline pysat/instruments/madrigal_pandas.py]: Docstring needs updating

File: pysat/instruments/madrigal_pandas.py [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+    Supports 'clean', 'dusty', 'dirty'
+    'Clean' is unknown for oblique modes, over 200 km for drifts
+    'Dusty' is unknown for oblique modes, over 200 km for drifts
+    'Dirty' is unknown for oblique modes, over 200 km for drifts
+    'None' None
+
+    Routine is called by pysat, and not by the end user directly.
+    
+    """
+    if self.clean_level in ['clean', 'dusty']:
```
Comment (aburrell) [inline pysat/instruments/madrigal_pandas.py]: Should also include "dirty" in this list.

Comment (rstoneback) [reply pysat/instruments/madrigal_pandas.py]: Updated

File: pysat/instruments/madrigal_pandas.py (hunk 4 above)
Comment (rstoneback) [reply pysat/instruments/madrigal_pandas.py]: Updated.

File: pysat/instruments/madrigal_pandas.py (hunk 3 above)
Comment (rstoneback) [reply pysat/instruments/madrigal_pandas.py]: updated

File: pysat/instruments/madrigal_pandas.py (hunk 2 above)
Comment (rstoneback) [reply pysat/instruments/madrigal_pandas.py]: It doesn't actually default to JRO Madrigal, as far as I can tell. I updated the comment.

File: pysat/instruments/jro_isr.py (hunk 1 above)
Comment (rstoneback) [reply pysat/instruments/jro_isr.py]: Thanks!

**Comment by rstoneback** [comment]
I updated the code per the review comments. Testing in progress. Tests will likely fail due to the pyglow and CDAWeb troubles but that is unrelated to the changes here. Once I verify no other tests fail, this will be ready to merge.

**Comment by rstoneback** [comment]
All test failures are expected and unrelated to this code. Ready to merge.

**Comment by aburrell** [comment]
I verified this.  Merging

**Pull request merged**